lib/spandx/cli/commands/scan.rb

```diff
@@ -4,51 +4,69 @@ module Spandx
   module Cli
     module Commands
       class Scan
-        attr_reader :scan_path, :spinner
+        attr_reader :scan_path
 
         def initialize(scan_path, options)
           @scan_path = ::Pathname.new(scan_path)
           @options = options
-          @spinner = options[:show_progress] ? ::Spandx::Core::Spinner.new : ::Spandx::Core::Spinner::NULL
           require(options[:require]) if options[:require]
         end
 
         def execute(output: $stdout)
-          report = ::Spandx::Core::Report.new
-          each_file do |file|
-            spinner.spin(file)
-            each_dependency_from(file) do |dependency|
-              spinner.spin(file)
-              report.add(dependency)
+          with_printer(output) do |printer|
+            with_thread_pool(size: thread_count) do |thread|
```

Comment:
> with_thread_pool will always yield once and we will enqueue task for each dependency ?

It's runs the `Plugin.enhance` step in a worker thread for each dependency that is discovered. This decouples the searching and parsing of dependencies from the plugins that enhance them and print to stdout. This allows for streaming results to the stdout stream as soon as they are ready rather than processing dependencies synchronously. As you requested I removed the usage of the ThreadPool until further notice.

> Why do we need thread pool now?

See above answer.